wip/integrationguide.md

```diff
@@ -0,0 +1,110 @@
+# Integration Guide:  Porting Mimi Models from v0.4.0 to v0.5.0
+
+## Overview
+
+The release of Mimi v0.5.0 is a breaking release, necessitating the adaptation of existing models' syntax and structure in order for those models to run on this new version.  This guide provides an overview of the steps required to get most models using the v0.4.0 API working with v0.5.0.  It is **not** a comprehensive review of all changes and new functionalities, but a guide to the minimum stpes required to port old models between versions.  For complete information on the new version and it's functionalities, see the full documentation.
+
+This guide is organized into six main sections, each descripting an independent set of changes that can be undertaken in any order desired.  For clarity, these sections echo the organization of the `userguide`.
+
+1) Defining components
+2) Constructing a model
+3) Running the model
+4) Accessing results
+5) Plotting
+6) Advanced topics
+
+## Defining Components
+
+The `run_timestep` function is now contained by the `@defcomp` macro, and takes the parameters `p, v, d, t`, referring to Parameters, Variables, and Dimensions of the component you defined.  The fourth argument is a Timestep, which represents which timestep the model is at each time the function gets called.  Similarly, the optional `init` function is also contained by `@defcomp`, and takes the parameters `p, v, d`.  Thus, as described in the user guide, definiing a single component is now done as follows:
+
+```julia
+@defcomp component1 begin
+
+    # First define the state this component will hold
+    savingsrate = Parameter()
+
+    # Second, define the (optional) init function for the component
+    function init(p, v, d)
+    end
+
```

Comment:
I think we need to firm up the semantics of `init()` vs `run_timestep()`. 

- Is `init` called _before_ any timesteps, or is it called on the first timestep? 
- If init is called on the first timestep, should `run_timestep` run only from the 2nd timestep forward?
- What if a component is run with an offset (i.e., it starts after the first global timestep)?

Currently, `init()` is called for all components before the timestep loop, which at least is easy to explain, but then `run_timestep()` needs to avoid redundant behavior in 1st real timestep. Perhaps we should start by enumerating the use cases?

Reply:
@rjplevin This is a good point.  I'm a novice on the `init()` function, haven't really used it or seen it used much, but we should probably discuss this further to firm it up in the documentation.  That wouldn't quite belong in this doc as much as the general documentation.  I'll make a note that we should discuss further with @davidanthoff.